examples/src/bin/sha.rs

```diff
@@ -1,36 +1,35 @@
 //! Demonstrates the use of the SHA peripheral and compares the speed of
 //! hardware-accelerated and pure software hashing.
 
+//% CHIPS: esp32 esp32c2 esp32c3 esp32c6 esp32h2 esp32s2 esp32s3
+
 #![no_std]
 #![no_main]
 
-use esp32_hal::{
-    clock::ClockControl,
+use esp_backtrace as _;
+use esp_hal::{
     peripherals::Peripherals,
     prelude::*,
     sha::{Sha, ShaMode},
-    xtensa_lx,
 };
-use esp_backtrace as _;
 use esp_println::println;
+use examples::cycles;
 use nb::block;
-use sha2::{Digest, Sha512};
+use sha2::{Digest, Sha256};
 
 #[entry]
 fn main() -> ! {
     let peripherals = Peripherals::take();
-    let system = peripherals.SYSTEM.split();
-    let _clocks = ClockControl::boot_defaults(system.clock_control).freeze();
 
     let source_data = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa".as_bytes();
     let mut remaining = source_data;
-    let mut hasher = Sha::new(peripherals.SHA, ShaMode::SHA512);
+    let mut hasher = Sha::new(peripherals.SHA, ShaMode::SHA256);
 
     // Short hashes can be created by decreasing the output buffer to the desired
     // length
     let mut output = [0u8; 64];
```

Comment:
```suggestion
    let mut output = [0u8; 32];
```

The `soft_result` used in the assert is 32 bytes